content/en/developers/dogstatsd/high_throughput.md

```diff
@@ -218,12 +222,72 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
+
+```bash
+sysctl net.unix.max_dgram_qlen
+```
+
+If the value is < 512, you can increase it to 512 or more using this command:
+
+```bash
+sysctl -w net.unix.max_dgram_qlen=512
+```
+
+Add the following configuration to `/etc/sysctl.conf` to make this change permanent:
+
+```conf
+net.unix.max_dgram_qlen = 512
+```
+
+In the same manner, the `net.core.wmem_max` could be incremented to 4MiB to improve
+client writing performances:
+
+```conf
+net.core.wmem_max = 4194304
+```
+
+Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number in `datadog.yaml`:
+
+```yaml
+dogstatsd_so_rcvbuf: 4194304
+```
+
+### Ensure proper packet sizes
+
+Sending packets with an adequate size to the Dogstatsd server within the Agent will avoid extra CPU usage. The latest versions of the official Dogstatsd clients send packets with a size optimized for performance. You can skip this section if you are using them.
+
+If the packets sent are too small, the Agent will have to pack several together in order to process them in batches later in the pipeline. The official Dogstatsd clients are capable of grouping metrics to have the best ratio of the number of metrics per packet.
+
+If the Dogstatsd clients properly send packets having a size close to `dogstatsd_buffer_size`, the Agent will perform better. Be careful: the packets must not be larger than this size, otherwise, the Agent won't be able to load them completely in the buffer and some of metrics will be malformed. Please use the corresponding configuration field in your Dogstatsd clients.
+
+Note for UDP: as the UDP packets are most of the time going through the Ethernet and IP layer, it is be better to avoid IP packets fragmentation by limiting the packet size to a value lower than a single Ethernet frame on your network. Most of the time, IPv4 networks are configured with a MTU of 1500 bytes, in this situation the packet size of sent packets should be limited to 1472.
+
+Note for UDS: for the best performances, UDS packet should have a size of 8192 bytes.
+
+### Limit the maximum memory usage of the Agent
+
+The Agent tries to absorb burst of metrics sent by the Dogstatsd clients but in order to do so, it needs to use memory. Even if this is for a short amount of time and even if this memory is quickly released to the OS, a spike happens and that could be an issue in containerized environments where limit on memory usage could evict pods or containers.
+
+This first thing to do to not have the Dogstatsd server within the agent to reach its maximum memory usage is to avoid sending metrics in bursts in your application.
+
+Another thing to look at to limit the maximum memory usage is to reduce the buffering. The main buffer of the Dogstatsd server within the Agent is configurable with the `dogstatsd_queue_size` field (since Datadog Agent 6.1.0), its default value of `1024` induces an approximate maximum memory usage of 768MB.
```

Comment:
```suggestion
Another thing to look at to limit the maximum memory usage is to reduce the buffering. The main buffer of the DogStatsD server within the Agent is configurable with the `dogstatsd_queue_size` field (since Datadog Agent 6.1.0), its default value of `1024` induces an approximate maximum memory usage of 768MB.
```